e-keithel-backend/app/controllers/application_controller.rb

```diff
@@ -3,4 +3,28 @@ class ApplicationController < ActionController::API
   def encode_token(payload)
     JWT.encode(payload, 'secret')
   end
+
+  def decode_token
+    auth_header = request.headers['Authorization']
+    if auth_header 
+      token = auth_header.split(' ').last
+      begin
+        JWT.decode(token, 'secret', true, algorithm: 'HS256')
+      rescue JWT::DecodeError
+        nil
+      end
+    end
+  end
+
+  def authorized_user
+    decoded_token = decode_token()
+    if decoded_token
+      user_id = decoded_token[0]['user_id']
+      @user = User.find_by(id: user_id)
+    end
+  end
+
+  def authorize
+    render json: {message: 'You need to be logged in'}, status: :unauthorized unless authorized_user
```

Comment:
Great work implementing the jwt authentication. Please give the sources for jwt resources with how you implement it here!